lib/chef/provider/package/yum.rb

```diff
@@ -61,6 +63,11 @@ def load_current_resource
           current_resource.package_name(new_resource.package_name)
           current_resource.version(get_current_versions)
 
+          if new_resource.actions.include?(:swap)
+            @swap_from_current_resource = Chef::Resource::YumPackage.new(new_resource.swap_from)
+            swap_from_current_resource.version(get_swap_from_current_version)
```

Comment:
you probably don't need to use a current_resource-looking-thing here.  you can't ever wire it up to reporting, or to load_current_value or to converge_if_changed, so the whole machinery around that abstraction can't really be used.  you probably just want a helper that returns only the `swap_from_version(package_name)`